_posts/2020-04-20-wgpu-web.md

```diff
@@ -0,0 +1,67 @@
+
+## Running wgpu-rs natively
+
+As we previously discussed in [The rise of wgpu](http://gfx-rs.github.io/2019/03/06/wgpu.html), the gfx team has been busy with development of `wgpu-native` (a C API which implements the [WebGPU specification](https://gpuweb.github.io/gpuweb/)) and `wgpu-rs` (an idiomatic Rust wrapper around `wgpu-native`).
+
+Even though these crates are still relatively new, it's been very exciting to watch people build all kinds of neat projects which use them, including a [GUI library](https://github.com/hecrj/iced), a [pixel editor](https://github.com/cloudhead/rx), [tutorials](https://sotrh.github.io/learn-wgpu/), [multiple](https://github.com/hecrj/coffee) [game](https://github.com/Ruddle/oxidator) [engines](https://github.com/StarArawn/harmony), a [creative-coding toolkit](https://github.com/nannou-org/nannou), a [sailing navigation application](https://github.com/Yatekii/sailor), a [Brawl character file parser](https://github.com/rukai/brawllib_rs), a [hardware-accelerated pixel buffer](https://github.com/parasyte/pixels), [Python bindings](https://github.com/almarklein/wgpu-py), a [Rust vangers clone](https://github.com/kvark/vange-rs), a [Chinese calligraphy app](https://apps.apple.com/us/app/id1492608770), [Scopes bindings](https://nest.pijul.com/porky11/wgpu), and many more interesting projects. There is even an [awesome-wgpu list](https://github.com/rofrol/awesome-wgpu) that the community has started in order to aggregate wgpu-related learning resources and examples in one place.
+
+Because `wgpu-rs` is cross-platform, all of these projects can run natively on top of Vulkan, Metal, DX12, and DX11. `wgpu-rs` internally selects the best graphics API for the user based on which graphics APIs are available at runtime. For older devices which don't support these modern APIs, limited GL support is also [actively being worked on](https://github.com/gfx-rs/wgpu/issues/450).
+
+## Running wgpu-rs on the web
+
+When we started the `wgpu-rs` crate, we wanted to eventually support two backends:
+
+- a "native" backend implemented with `wgpu-native`/`gfx`
+- a "web" backend which uses the WebGPU API provided by the browser
+
+Until recently, `wgpu-rs` was only able to run on the native backend, simply because the WebGPU API wasn't available in any browsers. But now that's changing, because browser vendors have been making [rapid progress on implementations of WebGPU](https://github.com/gpuweb/gpuweb/wiki/Implementation-Status#implementation-status).
+
+With the WebGPU API beginning to become available in Firefox Nightly and Chrome Canary behind experimental flags, we decided it was time to add a web backend. We [mapped the existing `wgpu-rs` API to the browser API](https://github.com/gfx-rs/wgpu-rs/pull/193), compiled our examples using the `wasm32-unknown-unknown` target, and were quickly able to run our examples on the web:
```

Comment:
can we add a clause saying that this is a big WIP and things may panic or fail to present the result?